Add husky pre-commit git hooks to lint staged files

lovincyrus opened the pull request:
This PR introduces a pre-commit script that runs against staged git files using [`lint-staged`](https://www.npmjs.com/package/lint-staged). A tiny DX improvement that prettier checks our staged files locally before they run in the prettier checks github action.

closes #5353

![CleanShot 2024-07-30 at 11 46 43@2x](https://github.com/user-attachments/assets/508320be-d95f-4ba3-8d93-3b2ee1f57c98)

ericpgreen2 commented:
Looks good, I just noticed a couple things in CI:
1. We should set `HUSKY=0` in CI to skip the Husky install. See: https://typicode.github.io/husky/how-to.html#ci-server-and-docker
2. We should add the husky files to `.prettierignore`

<img width="1277" alt="image" src="https://github.com/user-attachments/assets/30a803d8-6e88-41d5-9b90-34618c5a0e02">

lovincyrus commented:
> Looks good, I just noticed a couple things in CI:
> 
> 1. We should set `HUSKY=0` in CI to skip the Husky install. See: https://typicode.github.io/husky/how-to.html#ci-server-and-docker
> 2. We should add the husky files to `.prettierignore`
> 
> <img alt="image" width="1277" src="https://private-user-images.githubusercontent.com/14206386/353665883-30a803d8-6e88-41d5-9b90-34618c5a0e02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI0NDgzMDEsIm5iZiI6MTcyMjQ0ODAwMSwicGF0aCI6Ii8xNDIwNjM4Ni8zNTM2NjU4ODMtMzBhODAzZDgtNmU4OC00MWQ1LTliOTAtMzQ2MThjNWEwZTAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzMxVDE3NDY0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA5ZjczOWIxNDIzY2Q3YjE1OWMwZGEwNjYwYWQ1ZGY1NmU2YWFhZDJlOTA1ODMxYWQ4ZmE4ZDMzZWY3NTQzMWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gfEdEYKQNYkDu1vFFXtExjUEYZN2tz-Kq2hydigNueU">

Nice call, addressed the suggestions.
